src/main/java/org/opensearch/knn/index/mapper/LuceneFieldMapper.java

```diff
@@ -0,0 +1,127 @@
+/*
+ * Copyright OpenSearch Contributors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package org.opensearch.knn.index.mapper;
+
+import com.google.common.collect.ImmutableMap;
+import lombok.AllArgsConstructor;
+import lombok.Getter;
+import lombok.NonNull;
+import org.apache.lucene.document.FieldType;
+import org.apache.lucene.document.KnnVectorField;
+import org.apache.lucene.document.StoredField;
+import org.apache.lucene.index.IndexOptions;
+import org.apache.lucene.index.VectorSimilarityFunction;
+import org.apache.lucene.index.VectorValues;
+import org.opensearch.common.Explicit;
+import org.opensearch.index.mapper.ParseContext;
+import org.opensearch.knn.index.KNNMethodContext;
+import org.opensearch.knn.index.SpaceType;
+
+import java.io.IOException;
+import java.util.Map;
+import java.util.Optional;
+
+import static org.opensearch.knn.index.SpaceType.COSINESIMIL;
+import static org.opensearch.knn.index.SpaceType.INNER_PRODUCT;
+import static org.opensearch.knn.index.SpaceType.L2;
+
+/**
+ * Field mapper for case when Lucene has been set as an engine.
+ */
+public class LuceneFieldMapper extends KNNVectorFieldMapper {
+
+    private static final int MAX_DIMENSION = VectorValues.MAX_DIMENSIONS;
```

Comment:
Why not use the constant directly? Any reason why we are storing a constant as another constant?

Reply:
I'm looking at it as an adapter, we use our constant in multiple places, and I'd like to minimize direct references to Lucene classes.

Reply:
You can always static import constant. if it is loaded once, jvm will reuse.

Reply:
I was thinking from the code organization point, not the runtime